### PR TITLE
Android components quarantine

### PR DESCRIPTION
**Pull request by Pike**

### CODE REVIEW

File: mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -25,4 +25,30 @@
     <string name="mozac_feature_prompts_allow">Allow</string>
     <!-- Text of a negative button in dialog requesting to open a new window. -->
     <string name="mozac_feature_prompts_deny">Deny</string>
+    <!-- Title of the month chooser dialog. -->
+    <string name="mozac_feature_prompts_set_month">Pick a month</string>
+    <!-- January (3 character long), used on the month chooser dialog. -->
+    <string name="mozac_feature_prompts_jan">Jan</string>
```
Comment (Delphine) [inline mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: @Pike Isn't this going to be complicated for many locales (ie, shortening name months to 3 characters)?
Took a look around at our other products, and I don't know if there is a workaround in this case

Comment (Pike) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: https://www.unicode.org/cldr/charts/35/by_type/date_&_time.gregorian.html#Months_-_abbreviated_-_Standalone has some bad counter examples, but yeah, this is something that we should confirm that it's going to work with "few characters, but maybe more than 3"

Comment (Delphine) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: @Amejia481 or @pocmo : Do you have any idea if we can have more than 3 characters for this? We'd need to update the localization comments if that's the case. Thanks!

Comment (Amejia481) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: @Delphine having fewer won't be a problem, but having more than five could cause some issues with the actual layout. I will update the comments and the layout to adapt independently of the amount of charters.

This is where we are using the string.
image

Comment (Amejia481) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: https://github.com/mozilla-mobile/android-components/issues/3975

Comment (pocmo) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: The same dialog should exist in Fennec - maybe we can compare how it was translated there?

Comment (Amejia481) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: @pocmo  they are using a deprecated version :(

Comment (Amejia481) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: [This is what they are using: ](https://dxr.mozilla.org/mozilla-central/source/mobile/android/base/java/org/mozilla/gecko/widget/DateTimePicker.java#685)  `DateUtils.getMonthString(int month, int abbrev) ` [it was deprecated in API level 17 ](https://developer.android.com/reference/android/text/format/DateUtils.html#getMonthString(int,%20int))

Comment (Pike) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: There might be a corresponding method in ICU now

Comment (Amejia481) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: With `ICU` do you mean [android.icu.text](https://developer.android.com/reference/android/icu/text/package-summary)?

Comment (Pike) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: Yeah, though I don't know exact APIs. It does have all the data you're interested in, I guess the question is if they shell out just the abbreviated month.

Comment (pocmo) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: The deprecation notice mentions `SimpleDateFormat`- is that not an option?

Comment (pocmo) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: Our own translations will definitely be handy once we do locale switching and have translations for less common languages.

Comment (Pike) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: It's worth testing, gecko for example ships icu datasets beyond the locales we ship

Comment (Amejia481) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: What you thing about using? We could build the list of month with the output :

```kotlin
    val today = Calendar.getInstance()
    for( month in 0..11){
        today.set(Calendar.MONTH, month)
        val formatter = SimpleDateFormat("MMM", Locale.getDefault())
        println(formatter.format(today.time))
    }
```

**Output**
```
Jan
Feb
Mar
Apr
May
Jun
Jul
Aug
Sep
Oct
Nov
Dec
```

Comment (Amejia481) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: **Locale.forLanguageTag("ES")**
```
ene
feb
mar
abr
may
jun
jul
ago
sep
oct
nov
dic
```

**Locale.GERMANY** 
```
Jan
Feb
Mär
Apr
Mai
Jun
Jul
Aug
Sep
Okt
Nov
Dez
```

**Locale.FRANCE**
```
janv.
févr.
mars
avr.
mai
juin
juil.
août
sept.
oct.
nov.
déc.
```

**Locale.KOREAN**
```
1월
2월
3월
4월
5월
6월
7월
8월
9월
10월
11월
12월
```

**Locale.CHINESE**
```
一月
二月
三月
四月
五月
六月
七月
八月
九月
十月
十一月
十二月
```

Comment (pocmo) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: Looks good. My favorite language for Focus was `fy_NL` - can it handle that? :D

Comment (Amejia481) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: When I used `Locale.forLanguageTag("fy_NL")`, I doesn't find a locale for that, I'm not sure if  `Locale.forLanguageTag("NL")` is an equivalent? 

**Locale.forLanguageTag("NL")**
```
jan
feb
mrt
apr
mei
jun
jul
aug
sep
okt
nov
dec
```

Comment (Amejia481) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: http://www.localeplanet.com/icu/fy-NL/index.html

Comment (Pike) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: That's `nl` for Dutch. The expected result should be https://github.com/unicode-org/cldr/blob/f99e5f01cb00c321ea2239d99c56923ded53f6ea/common/main/fy.xml#L2084-L2097.

Do we know what https://developer.android.com/reference/android/icu/text/DateFormat.html#getAvailableLocales() returns?

Comment (pocmo) [reply mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml]: This table seems to be interesting too, our min. SDK version of 21 supports ICU 53 / CLDR 25:
https://developer.android.com/guide/topics/resources/internationalization

**Comment by Amejia481** [comment]
> Do we know what https://developer.android.com/reference/android/icu/text/DateFormat.html#getAvailableLocales() returns?


`DateFormat.html#getAvailableLocales() `Added in API level 24

"af"
"af_NA"
"af_ZA"
"agq"
"agq_CM"
"ak"
"ak_GH"
"am"
"am_ET"
"ar"
 "ar_001"
 "ar_AE"
 "ar_BH"
 "ar_DJ"
 "ar_DZ"
 "ar_EG"
 "ar_EH"
 "ar_ER"
 "ar_IL"
 "ar_IQ"
 "ar_JO"
 "ar_KM"
 "ar_KW"
 "ar_LB"
 "ar_LY"
 "ar_MA"
 "ar_MR"
 "ar_OM"
 "ar_PS"
 "ar_QA"
 "ar_SA"
 "ar_SD"
 "ar_SO"
 "ar_SS"
 "ar_SY"
 "ar_TD"
 "ar_TN"
 "ar_XB"
 "ar_YE"
 "as"
 "as_IN"
 "asa"
 "asa_TZ"
 "ast"
 "ast_ES"
 "az"
 "az__#Cyrl"
 "az_AZ_#Cyrl"
 "az__#Latn"
 "az_AZ_#Latn"
 "bas"
 "bas_CM"
 "be"
 "be_BY"
 "bem"
 "bem_ZM"
 "bez"
 "bez_TZ"
 "bg"
 "bg_BG"
 "bm"
 "bm_ML"
 "bn"
 "bn_BD"
 "bn_IN"
 "bo"
 "bo_CN"
 "bo_IN"
 "br"
 "br_FR"
 "brx"
 "brx_IN"
 "bs"
 "bs__#Cyrl"
 "bs_BA_#Cyrl"
 "bs__#Latn"
 "bs_BA_#Latn"
 "ca"
 "ca_AD"
 "ca_ES"
 "ca_FR"
 "ca_IT"
 "ccp"
 "ccp_BD"
 "ccp_IN"
 "ce"
 "ce_RU"
 "cgg"
 "cgg_UG"
 "chr"
 "chr_US"
 "ckb"
 "ckb_IQ"
 "ckb_IR"
 "cs"
 "cs_CZ"
 "cy"
 "cy_GB"
 "da"
 "da_DK"

**Comment by Delphine** [comment]
Not sure my comment ever got posted or if I just accidentally deleted it - so please bear with me.
I admit I'm a bit confused on where this is at at this point :)
It seems like the localization comment still needs to be updated. Not sure if there's anything else pending? We're blocked on exposing these strings until we get a fix. thanks!

**Comment by Amejia481** [comment]
@Pike @pocmo To fix this issue, which approach should we take? (We mentioned many, but we never agreed on one)

**Comment by Pike** [comment]
Let's just keep those 12 strings, it's not that much work.

I think we should replace "3 characters" with "short" in the comments, as we plan to soften that requirement in https://github.com/mozilla-mobile/android-components/issues/3975.

**Comment by Amejia481** [comment]
I opened a PR on the AC repository https://github.com/mozilla-mobile/android-components/pull/4134

**Pull request merged**